apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/v1/controller/ReleaseController.java

```diff
@@ -71,4 +80,72 @@ public OpenReleaseDTO loadLatestActiveRelease(@PathVariable String appId, @PathV
     return OpenApiBeanUtils.transformFromReleaseDTO(releaseDTO);
   }
 
+  @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/operator/{operator}/releases",
+          method = RequestMethod.POST)
+  public ReleaseDTO createGrayRelease(@PathVariable String appId,
+                                      @PathVariable String env, @PathVariable String clusterName,
+                                      @PathVariable String namespaceName, @PathVariable String branchName,
+                                      @PathVariable String operator,
+                                      HttpServletRequest request) {
+
+    NamespaceReleaseModel model = new NamespaceReleaseModel();
+    model.setAppId(appId);
+    model.setEnv(env.toUpperCase());
+    model.setClusterName(branchName);
+    model.setNamespaceName(namespaceName);
+    model.setReleasedBy(operator);
+
+    if (model.isEmergencyPublish() && !portalConfig.isEmergencyPublishAllowed(Env.valueOf(env.toUpperCase()))) {
+      throw new BadRequestException(String.format("Env: %s is not supported emergency publish now", env));
+    }
+
+    ReleaseDTO createdRelease = releaseService.publish(model);
+
+    ConfigPublishEvent event = ConfigPublishEvent.instance();
+    event.withAppId(appId)
+            .withCluster(clusterName)
+            .withNamespace(namespaceName)
+            .withReleaseId(createdRelease.getId())
+            .setGrayPublishEvent(true)
+            .setEnv(Env.valueOf(env.toUpperCase()));
+
+    publisher.publishEvent(event);
+
+    return createdRelease;
+  }
+
+  @RequestMapping(value = "/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/branches/{branchName}/operator/{operator}/gray-del-releases",
+          method = RequestMethod.POST)
+  public ReleaseDTO createGrayDelRelease(@PathVariable String appId,
+                                         @PathVariable String env, @PathVariable String clusterName,
+                                         @PathVariable String namespaceName, @PathVariable String branchName,
+                                         @PathVariable String operator,
```

Comment:
normally, operator should be a RequestParam(query parameter) instead of a PathVariable